### PR TITLE
Add new make_fun3 instruction

### PR DESCRIPTION
**Pull request by sverker**
Compared to old make_fun2:
- No GC done.
- Environment and destination are arguments of instruction and not assumed x-registers.

Results in more effective register allocations.

### CODE REVIEW

**Comment by bjorng** [comment]
Looks good to me.

I have only one minor nit to pick: For `Co-authored-by` to work as intended, it must be placed last in the commit message.

File: lib/compiler/src/genop.tab [hunk 1]
```diff
@@ -608,3 +608,5 @@ BEAM_FORMAT_NUMBER=0
 ##       it will never fail at runtime, or 'resume' when we know the input is
 ##       a match context.
 170: bs_start_match4/4
+
+171: make_fun3/3
```
Comment (bjorng) [inline lib/compiler/src/genop.tab:612]: Please document make_fun/3.

**Comment by bjorng** [comment]
@sverker Two test cases fail in the compiler test suite on host ecthelion (to recreate locally, give the `cover` option when runnning the test suites).

You will need to include `no_make_fun3` must be included in `test_lib:opt_opts/1`.

File: lib/compiler/src/genop.tab (hunk 1 above)
Comment (sverker) [reply lib/compiler/src/genop.tab:617]: done

**Comment by sverker** [comment]
*** MERGED *** MERGED *** MERGED ***
This PR is merged to master at 1073fa2

**Pull request closed without merging**